rasterio/features.py

```diff
@@ -335,11 +335,11 @@ def bounds(geometry, north_up=True):
         return tuple(geometry['bbox'])
 
     geom = geometry.get('geometry') or geometry
-    return _bounds(geom, north_up=north_up)
+    return _bounds(geom, north_up=north_up, transform=transform)
 
 
 def geometry_window(dataset, shapes, pad_x=0, pad_y=0, north_up=True,
-                    pixel_precision=3):
+                    rotated=False, pixel_precision=3):
```

Comment:
@whlteXbread what would you think about detecting the need to rotate from the dataset's transform within this function instead of in `mask()`? Passing `rotated=False` accidentally when the dataset *is* rotated will always give an erroneous result, no? And we might be able to prevent that.